pluto.jl

```diff
@@ -126,26 +126,40 @@ $$[CPU] ↔ [RAM] ↔ [硬盘缓存] ↔ [硬盘]$$
 
 # ╔═╡ f58d428c-8aef-11eb-3127-89d729e23823
 md"""
-## Avoid cache misses
-The RAM is faster than the disk, and the CPU in turn is faster than RAM. A CPU ticks like a clock, with a speed of about 3 GHz, i.e. 3 billion ticks per second. One "tick" of this clock is called a *clock cycle*. While this is a simplification, you may imagine that every cycle, the CPU executes a single, simple command called a *CPU instruction* which does one operation on a small piece of data. The clock speed then can serve as a reference for other timings in a computer. It is worth realizing just how quick a clock cycle is: In one cycle, a photon will travel only around 10 cm. In fact, modern CPUs are so fast that a significant constraint on their physical layout is that one must take into account the time needed for electricity to move through the wires inside them, so called wire delays.
+## 避免缓存未命中
+RAM 比内存快，而 CPU 又比 RAM 快。CPU滴答类似于时钟，速度约为 3 GHz，即每秒滴答30亿次。
+CPU 时钟的一次**滴答**被称为**时钟周期**。
+虽然这做了简化，但是你可以想象，在每个周期内，CPU都会执行一条简单的**指令**，该指令执行一次对小块数据的操作。
+时钟速度可以作为计算机中其他计时器的参考。
+了解时钟周期到底有多快是很值得的：光在一个时钟周期内只前进 10 cm。
+事实上，现代 CPU 如此之快，以至于一个约束物理布局的重要因素是必须考虑电子在内部线路传输所花的时间，这被称为线路延迟。
 
-On this scale, reading from RAM takes around 500 clock cycles. Similarly to how the high latency of disks can be mitigated by copying data to the faster RAM, data from RAM is copied to a smaller memory chip physically on the CPU, called a *cache*. The cache is faster because it is physically on the CPU chip (reducing wire delays), and because it uses a faster type of storage, static RAM, instead of the slower (but cheaper to manufacture) dynamic RAM which is what the main RAM is made of. Because the cache must be placed on the CPU, limiting its size, and because it is more expensive to produce, a typical CPU cache only contains around $10^8$ bits, around 1000 times less than RAM. There are actually multiple layers of CPU cache, but here we simplify it and just refer to "the cache" as one single thing:
+在该尺度上，从 RAM 中读取数据需要花费 500 个时钟周期。
+通过将数据拷贝到 RAM 可以减缓硬盘延迟。与之类似，RAM 上的数据也可以拷贝到 CPU 芯片上的一小块内存，称作**缓存（cache）**。
+缓存更快的原因是，它物理上位于CPU芯片（减少线路延迟），同时它使用一类更快的存储方式，静态RAM，而不是主流 RAM 所使用的动态 RAM方案。
```

Comment:
这里 RAM 有时指内存，有时有泛指 random access memory 技术...

```suggestion
缓存比内存更快的原因有两点：一方面在于缓存本身位于 CPU 内部因此线路延迟是很低的，另一方面在于 CPU 缓存一般采用静态 RAM 这种更快的存储方案，而不是主流内存中所采用的便宜的动态 RAM 的方案。
```

Reply:
静态 RAM 应该就是泛指，但我想或者不用特别强调？

Reply:
但.. "主流 RAM 所使用的动态 RAM方案" 里的第一个 RAM 指的是内存.

Reply:
> 但.. "主流 RAM 所使用的动态 RAM方案" 里的第一个 RAM 其实指的是内存？

啊是。第一个RAM指的是随机存取存储器，而第二个是存储器的技术。所以我加了方案二字以区分。